sonar-db/src/main/java/org/sonar/db/version/CreateTableBuilder.java

```diff
@@ -0,0 +1,280 @@
+      "There can't be more than one auto increment column");
+  }
+
+  public CreateTableBuilder withPkConstraintName(String pkConstraintName) {
+    this.pkConstraintName = checkDbIdentifier(pkConstraintName, "Primary key constraint name", CONSTRAINT_NAME_MAX_SIZE);
+    return this;
+  }
+
+  private String createTableStatement() {
+    StringBuilder res = new StringBuilder("CREATE TABLE ");
+    res.append(tableName);
+    res.append(" (");
+    appendPkColumns(res);
+    appendColumns(res, dialect, columnDefs);
+    appendPkConstraint(res);
+    res.append(')');
+    appendLOBStorageClause(res, dialect, columnDefs);
+    appendCollationClause(res, dialect);
+    return res.toString();
+  }
+
+  private void appendPkColumns(StringBuilder res) {
+    appendColumns(res, dialect, pkColumnDefs);
+    if (!pkColumnDefs.isEmpty() && !columnDefs.isEmpty()) {
+      res.append(',');
+    }
+  }
+
+  private void appendColumns(StringBuilder res, Dialect dialect, List<ColumnDef> columnDefs) {
+    if (columnDefs.isEmpty()) {
+      return;
+    }
+    Iterator<ColumnDef> columnDefIterator = columnDefs.iterator();
+    while (columnDefIterator.hasNext()) {
+      ColumnDef columnDef = columnDefIterator.next();
+      res.append(columnDef.getName());
+      res.append(' ');
+      appendDataType(res, dialect, columnDef);
+      appendNullConstraint(res, columnDef);
+      appendColumnFlags(res, dialect, columnDef);
+      if (columnDefIterator.hasNext()) {
+        res.append(',');
+      }
+    }
+  }
+
+  private void appendDataType(StringBuilder res, Dialect dialect, ColumnDef columnDef) {
+    if (PostgreSql.ID.equals(dialect.getId()) && isAutoIncrement(columnDef)) {
+      if (columnDef instanceof BigIntegerColumnDef) {
+        res.append("BIGSERIAL");
+      } else if (columnDef instanceof IntegerColumnDef) {
+        res.append("SERIAL");
+      } else {
+        throw new IllegalStateException("Column with autoincrement is neither BigInteger nor Integer");
+      }
+    } else {
+      res.append(columnDef.generateSqlType(dialect));
+    }
+  }
+
+  private boolean isAutoIncrement(ColumnDef columnDef) {
+    Collection<ColumnFlag> columnFlags = this.flagsByColumn.get(columnDef);
+    return columnFlags != null && columnFlags.contains(ColumnFlag.AUTO_INCREMENT);
+  }
+
+  private static void appendNullConstraint(StringBuilder res, ColumnDef columnDef) {
+    if (columnDef.isNullable()) {
+      res.append(" NULL");
+    } else {
+      res.append(" NOT NULL");
+    }
+  }
+
+  private void appendColumnFlags(StringBuilder res, Dialect dialect, ColumnDef columnDef) {
+    Collection<ColumnFlag> columnFlags = this.flagsByColumn.get(columnDef);
+    if (columnFlags != null && columnFlags.contains(ColumnFlag.AUTO_INCREMENT)) {
+      switch (dialect.getId()) {
+        case Oracle.ID:
+          // no auto increment on Oracle, must use a sequence
+          break;
+        case PostgreSql.ID:
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This case's code block is the same as the block for the case on line 178. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1871)